src/common/transformations/include/ov_ops/type_relaxed.hpp

```diff
@@ -161,9 +161,10 @@ class TemporaryReplaceOutputType {
     TemporaryReplaceOutputType(Output<Node> output, element::Type tmp_type) : m_output(output) {
         // save original element type in order to restore it in the destructor
         orig_type = m_output.get_element_type();
-        OPENVINO_SUPPRESS_DEPRECATED_START
-        m_output.get_tensor().set_element_type(tmp_type);
-        OPENVINO_SUPPRESS_DEPRECATED_END
+        if (auto parameter = dynamic_cast<ov::op::v0::Parameter*>(m_output.get_node())) {
+            parameter->set_element_type(tmp_type);
+            parameter->validate_and_infer_types();
+        }
```

Comment:
This change make a test failed, seems like the set type doesn't work for output node, @ilya-lavrenov  @vurusovs can u help take a look or any suggestions?
![image](https://github.com/openvinotoolkit/openvino/assets/52686861/029ed87d-39e9-43b6-a824-cf7243bd8992)

Reply:
Looks like transformation should work not only for parameters. Let's try to set new tensor with desired type and inherited shape via `m_output.set_tensor_ptr()`

Reply:
> Let's try to set new tensor with desired type and inherited shape via m_output.set_tensor_ptr()

Maybe it's better to remove deprecation then?